protocols.go

```diff
@@ -24,6 +24,7 @@ const (
 	P_UNIX              = 400
 	P_P2P               = 421
 	P_IPFS              = P_P2P // alias for backwards compatibility
+	P_HOST_HEADER       = 476
```

Comment:
476 is to be before both HTTP and WS (which works over HTTP and is 477).
I have no strong opinion about the name, http_host_header works as well. Once agreed with @MarcoPolo or other multiformats maintainers, I update this PR to the final name.

Reply:
I'd also prefer 481 so that it's right after HTTP.

Slight preference for `host-header` @marten-seemann because it looks nicer in text:
```
/ip4/1.2.3.4/tcp/80/http/host-header/example.com
vs
/ip4/1.2.3.4/tcp/80/http/http-host-header/example.com
```

I recognize this is a stylistic choice and this doesn't affect over the wire representation. So no strong opinion here. Do you feel strongly for `http-host-header` @marten-seemann ?

Reply:
updated to 481. still holding on changing the name atm.

Reply:
> Slight preference for host-header @marten-seemann because it looks nicer in text

Good point! I think this changes my preference from being slightly in favor of HTTP_HOST_HEADER to being slightly in favor of HOST_HEADER.